### DISCUSSION TITLE
ADC

### DISCUSSION THREAD

**Discussion by JingtaiChen** [stm32duino/Arduino_Core_STM32#1816] [Q&A] [answered]
为什么ADC速度这么慢

**Comment by JingtaiChen** [comment]
我是用的STM32F103ZET6的开发板，ADC速度最快才能到1KHZ，而第三方的STM32duino采样速度能到170Khz，有什么办法可以提升ADC速度，有改变ADC速度的函数吗？I am using the STM32F103ZET6 development board. The fastest ADC speed can reach 1KHZ, while the third-party STM32duino sampling speed can reach 170Khz. Is there any way to improve the ADC speed? Is there a function to change the ADC speed?

**Comment by fpistm** [comment, marked as answer]
Hi @CJT1111 
Right ADC implementation to be compatible with Arduino API could not reach this speed.
Anyway you can disabled this implementation and do your own with HAL/LL.
Here an example of a Cube project converted to Arduino which use DMA:
https://www.stm32duino.com/viewtopic.php?f=41&t=110

**Comment by JingtaiChen** [comment]
好的，我试验试验。感谢您的回复OK, I'll try. Thank you for your reply
